src/json.rs

```diff
@@ -54,7 +54,9 @@ impl JsonValue {
     pub fn u128(&self, key: &str) -> Result<u128, JsonError> {
         match self {
             JsonValue::Object(o) => match o.get(key).ok_or(JsonError::MissingValue)? {
-                JsonValue::Number(n) => Ok(*n as u128),
```

Comment:
It's still seems strange to me that we do not accept numbers; only strings. What would be the issue with accepting both?

Reply:
I don't think that we need to accept numbers for `u128` as the `Number` is limited to `u64`. So we will have some undefined behaviour and thus it will be hard to track such use cases and get why do we have some errors in future.
When we limit it to strings we allow using only this invariant so we won't have overflow use cases.
Otherwise, we need to remember and choose if we are going to use `Number` or `String` for `u128`, for each use case, etc.

Reply:
There is not undefined behaviour, it's that the map only goes one way. So any `JsonValue::Number` can be converted safely into a `u128`, but not the other way around. But this function is only `JsonValue -> u128`, so supporting the `Number` variant is logically sound.

If someone tried to pass some json like `{ "amount" : 1_000_000_000_000_000_000_000 }` I would expect the error to be in the parsing step because the given number does not fit in a `u64`.

As is, I think people may get confused when then pass json like `{ "amount" : 10 }` and get back `ERR_FAILED_PARSE_U128` when `10` is a valid `u128`.

That said, after laying it out like this it would maybe be a little awkward to accept both numbers and strings for a field because then the schema for the API is a little complex (e.g. `amount: Number | String`). So I would be ok with enforcing the conditions that all numbers which _could_ be bigger than `u64::MAX` must be given as strings regardless of their size. Though in this case, I would say that we should distinguish a type error from a value error. I.e. we should have a separate error variant which can flag to the user they passed a number where they should have passed a string.

Reply:
I think we shouldn't allow using both numbers and strings. Nevertheless, I agree with you that we need to avoid confusion, that's exactly what I meant. Added another `JsonError` and test in 03ad248

Reply:
@birchmd  If we should put u128 number, it always should be a String. Also, we return u128 as String. So it's a completely consistent approach.